### PR TITLE
Support custom domains hosts

### PR DESCRIPTION
**Pull request by alvaro-shopify**
### WHY are these changes introduced?
Some teams want to run CLI3 inside a spin instance without using `ngrok` nor any other tunnelling solution. To achieve that an `nginx` configuration is going to be added to the partners constellation [here](https://github.com/Shopify/partners/pull/42756). Once merged, the requests to access the CLI should point to `cli.{SPIN_FQDN}`.
Right now, the template is not supporting any custom host except for `ngrok`
An external way to set a custom value for the setting `hosts` must be added

### WHAT is this pull request doing?
- The env variable `SHOP_CUSTOM_DOMAIN` added to support custom store domains is reused [here](https://github.com/Shopify/shopify-app-template-ruby/pull/44) for setting the `hosts`
- The value of the  `SHOP_CUSTOM_DOMAIN` env variable always contains the prefix `shopify` prepending the real domain.
- The prefix must be removed before setting the `hosts` value.
- The decision of reusing the variable is for not adding a new one specific for the Ruby template

## Checklist

**Note**: once this PR is merged, it becomes a new release for this template.

- [ ] I have added/updated tests for this change
- [ ] I have made changes to the `README.md` file and other related documentation, if applicable

### CODE REVIEW

**Comment by nelsonwittwer** [comment]
We've recently added the ability to add `host` as a [configuration option for the ruby api](https://github.com/Shopify/shopify-api-ruby/blob/main/lib/shopify_api/context.rb#L54). 

I'd love to see if you can use your custom domain with that option. I'd hope that it would work for you the way you are expecting. If it doesn't, I'd love to see if we can rewire the gem to use that config instead of introducing the concept of a new host configuration to keep things simple.

**Comment by alvaro-shopify** [comment]
Hi @nelsonwittwer. Thank you for your feedback. We didn't realize that we were already setting the value `ENV["HOST"]`, however it seems that is not affecting the value of `config.host`
Anyway, we have modified the PR to use the variable you pointed out which seems to make more sense and make the code looking really cleaner. Thanks

File: web/config/environments/development.rb [hunk 1]
```diff
@@ -8,6 +8,8 @@
   rescue
     []
   end << /[-\w.]+\.ngrok\.io/
+
+  config.hosts << URI(ENV.fetch("HOST", "")).host if ENV.fetch("HOST", "").present?
```
[the remark this answers is not in the document]
Comment (alvaro-shopify) [reply web/config/environments/development.rb:12]: @nelsonwittwer , do you know if we are doing something wrong?, we this change the app is working but we don't know if it is the best approach as you mention in your comments. Thanks in advance

[the remark this answers is not in the document]
Comment (nelsonwittwer) [reply web/config/environments/development.rb:12]: I must be missing some piece of context with how the template works with `config.hosts`, as long as it is working for you and we are setting the `HOST` ENV variable then we should be good to go

**Comment by alvaro-shopify** [comment]
Bumped, tested and it seems to work properly.

**Pull request merged**